streams/src/main/java/org/apache/kafka/streams/kstream/internals/SessionWindowedKStreamImpl.java

```diff
@@ -92,16 +89,17 @@
         }
 
         return aggregateBuilder.build(
+            AGGREGATE_NAME,
+            materialize(materializedInternal),
             new KStreamSessionWindowAggregate<>(
-                windows, materializedInternal.storeName(),
+                windows,
+                materializedInternal.storeName(),
                 aggregateBuilder.countInitializer,
                 aggregateBuilder.countAggregator,
-                countMerger
-            ),
-            AGGREGATE_NAME,
-            materialize(materializedInternal),
-            materializedInternal.isQueryable()
-        );
+                countMerger),
+            materializedInternal.isQueryable(),
+            materializedInternal.keySerde() != null ? new WindowedSerdes.SessionWindowedSerde<>(materializedInternal.keySerde()) : null,
```

Comment:
This is a bit tricky: For windowed aggregates, users are only required to pass in the "inner" serdes in materialized, so we can still preserve it but need to wrap it with the windowed serdes. Ditto else where.

Reply:
Yes this is safe. See my comment above.